test/acceptance/service-broker-api.instances.virtualHost.spec.js

```diff
@@ -0,0 +1,180 @@
+      const container = virtualHostStore.containerName;
+      const data = {
+        instance_guid: instance_id,
+        deployment_name: deployment_name
+      };
+      const filename = `virtual_hosts/${instance_id}/${instance_id}.json`;
+      const pathname = `/${container}/${filename}`;
+      Promise.onPossiblyUnhandledRejection(() => {});
+
+      before(function () {
+        virtualHostStore.cloudProvider = new lib.iaas.CloudProviderClient(config.virtual_host.provider);
+        mocks.cloudProvider.auth();
+        mocks.cloudProvider.getContainer(container);
+        return mocks.setup([
+          virtualHostStore.cloudProvider.getContainer()
+        ]);
+      });
+
+      afterEach(function () {
+        mocks.reset();
+      });
+
+      describe('#provision', function () {
+        it('returns 201 created', function () {
+          mocks.director.getDeploymentManifest();
+          mocks.cloudController.getServiceInstancesInSpaceWithName(instance_name, space_guid, true);
+          mocks.agent.getInfo();
+          mocks.virtualHostAgent.createVirtualHost(instance_id);
+          mocks.cloudProvider.upload(pathname, () => {
+            return true;
+          });
+          mocks.cloudProvider.headObject(pathname);
+          return chai.request(app)
+            .put(`${base_url}/service_instances/${instance_id}`)
+            .set('X-Broker-API-Version', api_version)
+            .auth(config.username, config.password)
+            .send({
+              service_id: service_id,
+              plan_id: plan_id,
+              organization_guid: organization_guid,
+              space_guid: space_guid,
+              parameters: parameters,
+              accepts_incomplete: accepts_incomplete
+            })
+            .then(res => {
+              expect(res).to.have.status(201);
+              expect(res.body.dashboard_url).to.equal(dashboard_url);
+              mocks.verify();
+            });
+        });
+        it('returns 404 not found when wrong service instance name is passed.', function () {
+          mocks.cloudController.getServiceInstancesInSpaceWithName(instance_name, space_guid, false);
+          return chai.request(app)
+            .put(`${base_url}/service_instances/${instance_id}`)
+            .set('X-Broker-API-Version', api_version)
+            .auth(config.username, config.password)
+            .send({
+              service_id: service_id,
+              plan_id: plan_id,
+              organization_guid: organization_guid,
+              space_guid: space_guid,
+              parameters: parameters,
+              accepts_incomplete: accepts_incomplete
+            })
+            .catch(res => {
+              expect(res).to.have.status(404);
+              mocks.verify();
+            });
+        });
+      });
+
+      describe('#deprovision', function () {
+        it('returns 200 OK', function () {
```

Comment:
can you add a use case for deprovision when instance is not found? It must return 410 Gone when instance or virtual host not found.

Reply:
added this use case.